### DISCUSSION TITLE
Read-only gallery initial import on powerful machine then move everything to a less powerful one?

### DISCUSSION THREAD

**Discussion by mck182** [immich-app/immich#4090] [Q&A]
I'd like to give the [Read-only gallery](https://documentation.immich.app/docs/features/read-only-gallery) a try, however I have a significant
amount of photos and videos - roughly 250k taking about 1.8TB.

I'm thinking I'd set immich up on a powerful machine to let it crunch
through all that faster and then move it all to a NAS, a comparatively
less performant machine just to serve those photos with maybe some
occasional additions.

This should work provided the mount paths and asset paths in each
container are the same on both machines right? Are the metadata
stored in the database? Meaning I could also dump the database and
import it on the second machine? What about the thumbnails? Are
they also stored in the database? Or one of the containers? I'm not
sure how to transfer those over. What other data would need to be
moved over for everything to work on the second machine without
needing to process everything again?

I'd appreciate any other insights into this you might have, thanks!

**Comment by ddshd** [comment]
You'd be basically following the steps on https://documentation.immich.app/docs/administration/backup-and-restore to move your instance from one machine to another. For read-only library the path to the files inside the container would have to stay the same (as you stated)

**Reply by mck182** [reply]
One thing that's not clear to me from that doc is where are all the
processed thumbnails and/or transcoded videos and how to move
those over, without the second machine needing to regenerate all
of that. I assume those are not stored in the database right?

Then at the bottom it talks about Filesystem and it does mention
that there is generated content, but it does not mention where is
that generated content stored. That's the part I'm after I think.
